external-crates/move/move-compiler/tests/sui_mode/entry_points/receiving_non_object_type.exp

```diff
@@ -0,0 +1,9 @@
+error[Sui E02002]: invalid 'entry' function signature
+  ┌─ tests/sui_mode/entry_points/receiving_non_object_type.move:6:25
+  │
+6 │     public entry fun no(_: Receiving<u64>) { abort 0 }
+  │            -----        ^  -------------- 'entry' parameters must be primitives (by-value), vectors of primitives, objects (by-reference or by-value), or vectors of objects
```

Comment:
Should this test be failing?

Reply:
I guess my point is, that `Receiving` should/will have `T: key`, so who cares here?

Reply:
This should be failing since we check if the type arg to `Receiving` is an object in the sui mode checks (since it's nonsense to have one that isn't an object type). I can relax that if we want.

To the `key` question -- `Receiving` should not and can not have `key` for a couple different reasons:
* From a semantics perspective `Receiving` must be ephemeral. In particular you shouldn't be able to `transfer` it or store it anywhere.
* `Receiving` should have `drop` unless we want to enforce that all `Receiving` object arguments have been received in the transaction. This then means we can't have a UID field and therefore it cannot be `key`.